sfge/src/main/java/com/salesforce/graph/ops/registry/Registry.java

```diff
@@ -0,0 +1,97 @@
+package com.salesforce.graph.ops.registry;
+
+import com.salesforce.graph.ops.expander.PathExpansionRegistry;
+import com.salesforce.graph.ops.expander.PathExpansionRegistryUtil;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * Registry framework can be used as an optimization mechanism to store smaller data in collections.
+ * If you find that an object is particularly large and requires many instances to be created and
+ * stored in collections: 1. Make the object implement {@link Indexable} and provide an Id form. 2.
+ * Extend {@link AbstractRegistryData} for this object type. 3. Extend {@link Registry} and provide
+ * a Supplier with a mapping of the object type to a method to create its {@link
+ * AbstractRegistryData} implementation. 4. Modify usages of the object type to store its {@link
+ * Long} Id in the required collections and look up from Registry wherever the object's instance
+ * value is required.
+ *
+ * <p>See {@link PathExpansionRegistry} and {@link PathExpansionRegistryUtil} for examples on how
+ * this framework can be used.
+ */
+public abstract class Registry {
+
+    protected final Map<Class, AbstractRegistryData> registryHolderMap;
+
+    public Registry() {
+        registryHolderMap = new HashMap<>();
+        for (Map.Entry<Class<? extends Indexable>, Supplier> entry :
+                this.getRegistrySupplier().entrySet()) {
+            registryHolderMap.put(entry.getKey(), (AbstractRegistryData) entry.getValue().get());
+        }
+    }
+
+    /**
+     * @return a Map of Class that implements {@link Indexable} to a Supplier to get an instance of
```

Comment:
Extremely minor gripe: "Mapping from ... to" instead of "of...to".

Reply:
Since this says "Map" and not "Mapping", isn't "Map of" correct?

Reply:
My mistake. I meant this should either be "mapping from...to" or "map of...by". "Map of...to" was confusing to me.